pkg/controllers/unsupported/unsupported_controller_test.go

```diff
@@ -63,19 +73,35 @@ var _ = Describe("CAPI unsupported controller", func() {
 				Name: capiClusterOperator.Name,
 			},
 		})
-
 		Expect(err).ToNot(HaveOccurred(), "should be able to reconcile the cluster-api ClusterOperator without erroring")
 
-		Eventually(func() (*configv1.ClusterOperator, error) {
-			err := cl.Get(ctx, capiClusterOperatorKey, capiClusterOperator)
-			return capiClusterOperator, err
-		}).Should(HaveField("Status.Conditions",
+		co := komega.Object(configv1resourcebuilder.ClusterOperator().WithName(controllers.ClusterOperatorName).Build())
+		Eventually(co).Should(HaveField("Status.Conditions",
 			SatisfyAll(
 				ContainElement(And(HaveField("Type", Equal(configv1.OperatorAvailable)), HaveField("Status", Equal(configv1.ConditionTrue)), HaveField("Message", Equal(capiUnsupportedPlatformMsg)))),
 				ContainElement(And(HaveField("Type", Equal(configv1.OperatorProgressing)), HaveField("Status", Equal(configv1.ConditionFalse)))),
 				ContainElement(And(HaveField("Type", Equal(configv1.OperatorDegraded)), HaveField("Status", Equal(configv1.ConditionFalse)))),
+				ContainElement(And(HaveField("Type", Equal(configv1.OperatorUpgradeable)), HaveField("Status", Equal(configv1.ConditionTrue)))),
 			),
 		), "should match the expected ClusterOperator status conditions")
+
+	})
+
+	It("should update the ClusterOperator status version to the desired one", func() {
```

Comment:
Do we want/have a test that changes the value to something incorrect, and then see if it gets update back to the correct thing?

Reply:
Yep, exactly

Reply:
Ah yes, we don't have that at the moment. Updated.
Also something to take into account is that this controller is going to disappear in the next PR #246 as it is going to get merged into a different one.
For that one I am planning more exhaustive integration tests and them starting from the manager rather than invoking the r.Reconcile() directly.
So any extra testing changes I'd suggest we could do there.